runtime/lua/vim/lsp/buf.lua

```diff
@@ -219,10 +227,14 @@ function M.format(options)
   end
 
   local function set_range(client, params)
-    if range then
-      local range_params =
-        util.make_given_range_params(range.start, range['end'], bufnr, client.offset_encoding)
-      params.range = range_params.range
+    local to_lsp_range = function(r)
+      return util.make_given_range_params(r.start, r['end'], bufnr, client.offset_encoding).range
+    end
+
+    if passed_multiple_ranges then
+      params.ranges = vim.tbl_map(to_lsp_range, range)
```

Comment:
Please add typing (`lsp.DocumentRangeFormattingParams` or `lsp.DocumentRangesFormattingParams`) to this local function.